### PR TITLE
Bump Spring Boot to 3.0.0-RC2

### PR DESCRIPTION
**Pull request by JohnNiang**
#### What type of PR is this?

/kind improvement
/area core
/milestone 2.0

#### What this PR does / why we need it:

Bump Spring Boot to 3.0.0-RC2. 

See https://github.com/spring-projects/spring-boot/releases/tag/v3.0.0-RC2 for more.

#### Does this PR introduce a user-facing change?

```release-note
升级 Spring Boot 至 3.0.0-RC2
```

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,6 +1,5 @@
 [*]
 charset = utf-8
-end_of_line = lf
```
Comment (JohnNiang) [inline .editorconfig:3]: Delete this config item for Windows developers.

**Review by ruibaby — COMMENTED** [review]
/approve

**Review by guqing — COMMENTED** [review]
/lgtm

**Pull request merged**